File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.java
class Solution {
    public int minimumDeletions(int[] nums) {
        int n=nums.length;
        int minindex=0;
        int maxindex=0;
        for(int i=0; i<n;i++)
        {
            if(nums[i]>nums[maxindex])
            {
                maxindex=i;
            }
            if(nums[i]<nums[minindex])
            {
                minindex=i;
            }
        }
        int l=Math.min(minindex,maxindex);
        int r=Math.max(minindex,maxindex);

        int f=r+1;
        int e=n-l;
        int both=l+1+n-r;
        return Math.min(f,Math.min(e,both));
        
    }
}

File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.py
class Solution(object):
    def minimumDeletions(self, nums):
        
        n = len(nums)

        min_index = 0
        max_index = 0

        for i in range(1, n):
            if nums[i] < nums[min_index]:
                min_index = i

            if nums[i] > nums[max_index]:
                max_index = i

        left = min(min_index, max_index)
        right = max(min_index, max_index)

        front = right + 1
        back = n - left
        both = (left + 1) + (n - right)

        return min(front, back, both)
        